geopandas/tools/overlay.py

```diff
@@ -176,3 +177,158 @@ def overlay(df1, df2, how, use_sindex=True):
 
     # Return geodataframe with new indices
     return GeoDataFrame(collection, index=range(len(collection)))
+
+def overlay_intersection(df1, df2):
+    ''''
+    Overlay Intersection operation used in overlay function
+    '''
+    # Spatial Index to create intersections
+    spatial_index = df2.sindex
+    df1['bbox'] = df1.geometry.apply(lambda x: x.bounds)
+    df1['sidx'] = df1.bbox.apply(lambda x:list(spatial_index.intersection(x)))
```

Comment:
the same for `df1['sidx']`. As far as I can see in the code, this is only used in itself (two lines below), and not in the actual merging? Then I would call it `sidx`